google/cloud/storage/internal/hash_function_impl.cc

```diff
@@ -29,8 +35,15 @@ namespace {
 using ::google::cloud::internal::InvalidArgumentError;
 using ::google::cloud::storage_internal::ExtendCrc32c;
 
-using ContextPtr = std::unique_ptr<EVP_MD_CTX, MD5HashFunction::ContextDeleter>;
+using ContextPtr = std::unique_ptr<void, MD5HashFunction::ContextDeleter>;
 
+#ifdef _WIN32
+ContextPtr CreateMD5HashCtx() {
+  BCRYPT_HASH_HANDLE hHash = nullptr;
+  BCryptCreateHash(BCRYPT_MD5_ALG_HANDLE, &hHash, nullptr, 0, nullptr, 0, 0);
+  return ContextPtr(hHash);
+}
+#else
```

Comment:
I can do that, but it will leak the OpenSSL headers to `hash_function_impl.h` (I could not find a way to forward-declare a C struct), which might be OK since the header is internal.

Reply:
Done. We leak the OpenSSL headers but that's what we've been doing before so there is no change to the status quo. And `hash_function_impl.h` gets included by few other places.